### ISSUE TITLE
`TimerRef` from `TimerWithThread::for_uuid_closures` is not cloneable

### ISSUE THREAD

**Issue by feelingsonice** [Bathtor/rust-hash-wheel-timer#6]
The `TimerRef` returned from `TimerWithThread::for_uuid_closures()` is not cloneable because `OneShotClosureState` and `PeriodicClosureState` are not cloneable. Even though `TimerRef` is:

```rust
#[derive(Clone)] // <-----------------
pub struct TimerRef<I, O, P>
where
    I: Hash + Clone + Eq,
    O: OneshotState<Id = I>,
    P: PeriodicState<Id = I>,
{
    work_queue: channel::Sender<TimerMsg<I, O, P>>,
}
```

A small inconvenience but this means some structuring limitations. For example, `TimerRef` implements `Timer` but `TimerWithThread` does not, so in practice I can't abstract the `Timer` implementation away using generics and have to keep the `TimerWithThread` around. Just a minor suggestion :)

**Comment by feelingsonice** [comment]
Took another look at the impl dont think it's possible since it's holding a closure.

**Comment by Bathtor** [comment]
Hmm, actually it's just holding the `channel::Sender`, no? Which should be cloneable. The rest is just part of the type definition of the messages carried on the channel. But that shouldn't affect whether the `TimerRef` is cloneable.
I'll try to look into it on the weekend.

**Comment by feelingsonice** [comment]
Yea that's what I thought too, hence why created this, but this:
```rust
use hierarchical_hash_wheel_timer::{thread_timer::TimerWithThread, ClosureTimer};
use tokio::time::sleep;

#[tokio::main]
async fn main() {
    let x = {
        let timer_core = TimerWithThread::for_uuid_closures();
        let mut timer_ref = timer_core.timer_ref();
        timer_ref.schedule_action_once(
            uuid::Uuid::new_v4(),
            std::time::Duration::from_secs(5),
            |_| {
                println!("Hello, world!");
            },
        );

        timer_ref.clone()
    };
    sleep(std::time::Duration::from_secs(10)).await;
}
```
gives the error:
```
error[E0599]: the method `clone` exists for struct `TimerRef<Uuid, OneShotClosureState<Uuid>, PeriodicClosureState<Uuid>>`, but its trait bounds were not satisfied
   --> src/main.rs:17:19
    |
17  |         timer_ref.clone()
    |                   ^^^^^ method cannot be called due to unsatisfied trait bounds
    |
   ::: /Users/bli/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hierarchical_hash_wheel_timer-1.2.0/src/timers.rs:174:1
    |
174 | pub struct OneShotClosureState<I> {
    | --------------------------------- doesn't satisfy `OneShotClosureState<Uuid>: Clone`
...
240 | pub struct PeriodicClosureState<I> {
    | ---------------------------------- doesn't satisfy `PeriodicClosureState<Uuid>: Clone`
    |
   ::: /Users/bli/.cargo/registry/src/index.crates.io-6f17d22bba15001f/hierarchical_hash_wheel_timer-1.2.0/src/thread_timer.rs:69:1
    |
69  | pub struct TimerRef<I, O, P>
    | ---------------------------- doesn't satisfy `_: Clone`
    |
    = note: the following trait bounds were not satisfied:
            `OneShotClosureState<Uuid>: Clone`
            which is required by `TimerRef<Uuid, OneShotClosureState<Uuid>, PeriodicClosureState<Uuid>>: Clone`
            `PeriodicClosureState<Uuid>: Clone`
            which is required by `TimerRef<Uuid, OneShotClosureState<Uuid>, PeriodicClosureState<Uuid>>: Clone`
```

**Comment by Bathtor** [comment]
Yeah, this is because I derive the `Clone` implementation, and Rust is very conservative and only implements traits if all generic types also implement it. But I can fix it by just implementing clone manually instead, as long as `channel::Sender` has the correct `Clone` implementation (i.e., also ignoring the type of the message).

**Comment by feelingsonice** [comment]
Thanks!

**Comment by Bathtor** [comment]
The fix is merged and released in version 1.3.0
